lib/trusted-apps.js

```diff
@@ -43,10 +43,19 @@ function middleware(req, res, next) {
   }
 
   let url = getRequestUrl(req);
-
-  let certificate = req.header(HEADER_CERTIFICATE);
-  let providerId = req.header(HEADER_PROVIDER_ID);
-  let signature = req.header(HEADER_SIGNATURE);
+  let certificate, providerId, signature;
+
+  if (typeof req.header === 'function') {
+    certificate = req.header(HEADER_CERTIFICATE);
+    providerId = req.header(HEADER_PROVIDER_ID);
+    signature = req.header(HEADER_SIGNATURE);
+  } else if (req.header && typeof req.header === 'object') {
+    certificate = req.header[HEADER_CERTIFICATE];
+    providerId = req.header[HEADER_PROVIDER_ID];
+    signature = req.header[HEADER_SIGNATURE];
+  } else {
+    return error(res, 'header-invalid-type', 'req.header is not a function or object');
```

Comment:
We could throw an exception here – this doesn't feel like a user-facing Unauthorized error so much as a development-time framework incompatibility

Reply:
Done it :)